File: flutter_plugin_cal/ios/Classes/FlutterPluginCalPlugin.m
#import "FlutterPluginCalPlugin.h"
#import <JavaScriptCore/JavaScriptCore.h>

@interface FlutterPluginCalPlugin ()

@property(nonatomic,strong)JSContext *context;

@end

@implementation FlutterPluginCalPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_plugin_cal"
            binaryMessenger:[registrar messenger]];
  FlutterPluginCalPlugin* instance = [[FlutterPluginCalPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  }else if([@"calculate" isEqualToString:call.method]){
      NSDictionary *arguments = [call arguments];
      NSString *resultText = arguments[@"infodata"];
      NSLog(@"打印传递过来的值====：%@",resultText);
      if (self.context == nil) {
          self.context = [[JSContext alloc] init];
      }
      if(!resultText){
          result(nil);
      }else{
          JSValue *value = [self.context evaluateScript:resultText];
          NSLog(@"计算结果====：%@",value);
          if(value == nil||[@"undefined" isEqualToString:value.toString]){
              result(nil);
          }else{
              result(value.toString);
          }
      }
  }else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: flutter_plugin_cal/android/src/main/java/com/zx/flutter_plugin_cal/FlutterPluginCalPlugin.java
package com.zx.flutter_plugin_cal;

import org.liquidplayer.webkit.javascriptcore.JSContext;
import org.liquidplayer.webkit.javascriptcore.JSValue;

import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** FlutterPluginCalPlugin */
public class FlutterPluginCalPlugin implements MethodCallHandler {
  private JSContext jsContext ;
  private FlutterPluginCalPlugin(){
    jsContext = new JSContext();
  }
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_plugin_cal");
    channel.setMethodCallHandler(new FlutterPluginCalPlugin());
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else if(call.method.equals("calculate")){
      Map arg = call.arguments();
      final String infoData = arg.get("infodata").toString();
      if(infoData.isEmpty()){
        result.success(null);
        return;
      }
      if(jsContext!=null){
        JSValue jsValue = null;
        try {
          jsValue = jsContext.evaluateScript(infoData);
        }catch (Exception e){
        }
        result.success(jsValue != null ? jsValue.toString() : null);
      }else {
        result.success(null);
      }
    }else {
      result.notImplemented();
    }
  }
}
